tools/src/h5diff/ph5diff_main.c

```diff
@@ -57,11 +62,6 @@ main(int argc, char *argv[])
     outBuffOffset = 0;
     g_Parallel    = 1;
 
-    MPI_Init(&argc, (char ***)&argv);
```

Comment:
Move the MPI_Init call up before anything else so that HDF5 can properly setup MPI attributes for cleanup on MPI_Finalize